### PR TITLE
feat(exhaustive-switch): add .ts file

### PR DESCRIPTION
**Pull request by rthomazel**
add .ts file

### CODE REVIEW

File: src/exhaustive-switch.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * This function enables the exhaustive check.
+ * It works by returning `never`, and accepting the first param as `never`
+ * @param _type The type that we're checking
+ * @param message A message to display if it errors
+ * @param lastWords A callback to call before the error
+ */
+export function assertUnreachable(
+  _type: never,
```
Comment (imteekay) [inline src/exhaustive-switch.ts]: I would only change this `_type` to `value` to not confuse the `type` and `value` spaces.

```suggestion
  value: never,
```

File: src/exhaustive-switch.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/**
+ * Here the compiler checks that we have handled all possible cases in the switch statement
+ * in other words, a compile-time exhaustive check
+ */
+
+/**
+ * This function enables the exhaustive check.
+ * It works by returning `never`, and accepting the first param as `never`
+ * @param _type The type that we're checking
```
Comment (imteekay) [inline src/exhaustive-switch.ts]: also here
```suggestion
 * @param value The type that we're checking
```

**Review by imteekay — COMMENTED** [review]
Great job! 🎉 

I added a comment to make the parameter more clear and I recommend using only single quotes. (I'll add eslint and prettier to the project, my bad!)

**Comment by rthomazel** [comment]
Nice, you're right, that's actually a `value` not a `type`

**Comment by imteekay** [comment]
Thanks for the contribution, @Thomazella!

**Pull request merged**